backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/process/C_OLCand_SetOverrideValues.java

```diff
@@ -56,12 +75,35 @@ protected void prepare()
 	{
 		final List<ProcessInfoParameter> parameterList = new ArrayList<>(getProcessInfo().getParameter());
 		final Map<String, ProcessInfoParameter> parameters = parameterList.stream().collect(Collectors.toMap(ProcessInfoParameter::getParameterName, param -> param));
+		final int bpartnerId = parameters.get(PARAM_BPartner).getParameterAsInt(-1);
+		if (bpartnerId > -1 && Check.isBlank(parameters.get(PARAM_Location).getParameterAsString()))
+		{
+			parameters.put(PARAM_Location, ProcessInfoParameter.of(PARAM_Location, getBPartnerLocationId(bpartnerId).getRepoId()));
+		}
+		params = new ProcessParams(ImmutableList.copyOf(parameters.values()));
+	}
 
-		if (Check.isNotBlank(parameters.get(PARAM_BPartner).getParameterAsString()) && Check.isBlank(parameters.get(PARAM_Location).getParameterAsString()))
+	private BPartnerLocationId getBPartnerLocationId(final int bpartnerId)
+	{
+		final Set<GLN> glns = createQuery().stream()
+				.map(record -> BPartnerLocationId.ofRepoId(record.getC_BPartner_ID(), record.getC_BPartner_Location_ID()))
```

Comment:
i think this creates an `n+1`problem.
To avoid it, you can use one of the `IQueryBuilder#andCollect*` methods and get all the bpartner-locations  right away.

Also note that when looking for the *target* bpartner-location (location with the same GLN, but of the overriding-bpartner), please don't consider deactivatedd ones

Reply:
rewritten to get everything in a single go.